src/main/java/org/vietabroader/controller/SpreadsheetConnectController.java

```diff
@@ -0,0 +1,50 @@
+package org.vietabroader.controller;
+
+import org.vietabroader.model.GlobalState;
+import org.vietabroader.model.VASpreadsheet;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SpreadsheetConnectController implements Controller {
+
+    private JButton btnConnect;
+    private JTextField txtConnect;
+    private static final Logger logger = LoggerFactory.getLogger(SpreadsheetConnectController.class);
+
+    public SpreadsheetConnectController setConnectButton(JButton btn) {
+        btnConnect = btn;
+        return this;
+    }
+
+    public SpreadsheetConnectController setTextSpreadsheetID(JTextField txt) {
+        txtConnect = txt;
+        return this;
+    }
+
+    @Override
+    public void control() {
+        btnConnect.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                GlobalState currentState = GlobalState.getInstance();
+                try {
```

Comment:
What I meant was the check for whether you are SIGNED_OUT should be removed, but you still need to wrap this code in the check of SIGNED_IN and CONNECTED status.